### PR TITLE
Updated deps to newest version and allow both react 16 and 17 as a peer #860

### PR DESCRIPTION
**Pull request by rkesters**
# Checklist

- [x] The code has been run through pretty `yarn run pretty`
- [x] The tests pass on CircleCI
- [x] You have referenced the issue(s) or other PR(s) this fixes/relates-to
- [x] The PR Template has been filled out (see below)
- [x] Had a beer/coffee because you are awesome

## What?
Update deps to support react 16 and react 17. Ensured the devDeps used compatible versions.

## Why?

- The package does not use anything that is specific to React 17, hence it compatiable with both React 16 and 17. 
- Out of the box the stroybook package would not run, this had to do we mismatched versions
- Use opportunity to update all deps to latest, no breaking changes

## How?
- made changes to package.json files
- updated 1 test file to support jest 27

### CODE REVIEW

**Comment by rkesters** [comment]
@dylanvorster please review.

File: diagrams-demo-gallery/package.json [hunk 1]
```diff
@@ -9,10 +9,10 @@
 		"url": "https://github.com/projectstorm/react-diagrams.git"
 	},
 	"scripts": {
-		"start": "../node_modules/.bin/start-storybook",
-		"storybook:build": "../node_modules/.bin/build-storybook -c .storybook -o .out",
-		"github": "../node_modules/.bin/storybook-to-ghpages",
-		"test:run": "../node_modules/.bin/jest --no-cache",
+		"start": "npx start-storybook",
```
Comment (dylanvorster) [inline diagrams-demo-gallery/package.json:12]: i would much prefer to use the binaries that are already inside the node_modules bin folder.

File: package.json [hunk 2]
```diff
@@ -39,65 +39,56 @@
...
-		"@storybook/theming": "^6.2.8",
-		"@types/dagre": "^0.7.44",
-		"@types/jest": "^26.0.22",
-		"@types/jest-environment-puppeteer": "^4.4.1",
-		"@types/lodash": "^4.14.168",
-		"@types/node": "^14.14.37",
-		"@types/promise": "^7.1.30",
-		"@types/puppeteer": "^5.4.3",
-		"@types/react": "^17.0.3",
+    "emotion": "11.*",
```
Comment (dylanvorster) [inline package.json:45]: i would much prefer it if u used a tool such as `ncu` to bump to exact exact versions

Comment (rkesters) [reply package.json:45]: Use a tool, i don't understand?

File: diagrams-demo-gallery/package.json (hunk 1 above)
Comment (rkesters) [reply diagrams-demo-gallery/package.json]: updated

File: package.json (hunk 2 above)
Comment (rkesters) [reply package.json]: updated to exact versions.

**Comment by rkesters** [comment]
@dylanvorster please review.

**Comment by dylanvorster** [comment]
I am going to rebump all packages though, I think peer deps can include 16, but dev version should be 17

**Pull request merged**